azure/petctl.py

```diff
@@ -0,0 +1,214 @@
+    )
+    
+     # ---------------------------------- #
+     #               SETUP                #
+     # ---------------------------------- #
+        
+    parser_setup = subparser.add_parser(
+        "setup", help="set up aks-engine, cluster and other dependencies"
+    )
+    
+    parser_setup.add_argument(
+        "--dns_prefix",
+        type=str,
+        required=False,
+        default = "azure-pytorch-elastic",
+        help="Dns prefix of the app",
+    )
+    
+    parser_setup.add_argument(
+        "--subscription_id",
+        type=str,
+        required=True,
+        help="Subscription id of the cluster",
+    )
+    
+    parser_setup.add_argument(
+        "--rg",
+        type=str,
+        required=True,
+        help="Resource group of the cluster",
+    )
+    
+    parser_setup.add_argument(
+        "--location",
+        type=str,
+        required=True,
+        help="Location of the cluster",
+    )
+    
+    parser_setup.add_argument(
+        "--client_id",
+        type=str,
+        required=True,
+        help="Service principal client id",
+    )
+    
+    parser_setup.add_argument(
+        "--client_secret",
+        type=str,
+        required=True,
+        help="Service Principal client secret",
+    )
+    
+    parser_setup.add_argument(
+        "--storage_account_name",
+        type=str,
+        required=True,
+        help="Pet blob storage secrets",
+    )
+    
+    parser_setup.add_argument(
+        "--storage_account_key",
+        type=str,
+        required=True,
+        help="Pet blob storage secrets",
+    )
+    
+    parser_setup.add_argument(
+        "--docker_server",
+        type=str,
+        required=True,
+        help="Docker server",
+    )
+    
+    parser_setup.add_argument(
+        "--docker_username",
+        type=str,
+        required=True,
+        help="Docker username",
+    )
+    
+    parser_setup.add_argument(
+        "--docker_password",
+        type=str,
+        required=True,
+        help="Docker password",
+    )
+    
+    parser_setup.set_defaults(func=setup)
+    
+     # ---------------------------------- #
+     #        CONFIGURE JOB YAML          #
+     # ---------------------------------- #
+    
+    parser_configure = subparser.add_parser(
+        "configure", help="Generate yaml job file"
+    )
+
+    parser_configure.add_argument("--name", required=True, help="config parameters")
+    parser_configure.add_argument(
+        "--min_size",
+        type=int,
+        required=False,
+        help="minimum number of worker hosts to continue training",
+    )
+    parser_configure.add_argument(
+        "--max_size",
+        type=int,
+        required=False,
+        help="maximum number of worker hosts to allow scaling out",
+    )
+    parser_configure.add_argument(
+        "--size",
+        type=int,
+        required=False,
+        help="number  of worker hosts to start the job with",
```

Comment:
The help text seems a bit confusing. Setting `size` instead of min/max size just means `min_size` = `max_size` = `size`.